core/src/main/java/com/dtolabs/rundeck/core/execution/script/ScriptfileUtils.java

```diff
@@ -149,6 +165,10 @@ private static void writeReader(
                 null == style ?
                         LineEndingStyle.LOCAL.getLineSeparator() :
                         style.getLineSeparator();
+
+        if(addBom && StandardCharsets.UTF_8.equals(writer.getEncoding()) && style == LineEndingStyle.WINDOWS) {
```

Comment:
strangely, this didn't work on macOS (writer.getEncoding() returns `UTF8` without dash) .
Forcing that comparison works for me